lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -726,11 +758,18 @@ class EditingState {
   /// The offset at which the text selection terminates.
   final int? extentOffset;
 
+  /// The offset at which [CompositionAwareMixin.composingText] begins, if any.
+  final int? composingBaseOffset;
+
+  /// The offset at which [CompositionAwareMixin.composingText] terminates, if any.
+  final int? composingExtentOffset;
+
   /// Whether the current editing state is valid or not.
   bool get isValid => baseOffset! >= 0 && extentOffset! >= 0;
 
   @override
-  int get hashCode => Object.hash(text, baseOffset, extentOffset);
+  int get hashCode => Object.hash(
+    text, baseOffset, extentOffset, composingBaseOffset, composingExtentOffset);
```

Comment:
Nit: I think the close parenthesis goes on the next line according to the styleguide.